test/integration/extension/lib/bigcommerce/CartRepositoryTest.js

```diff
@@ -63,8 +76,38 @@ describe('BigCommerceCartRepository - integration', () => {
     return subjectUnderTest.destroy().should.eventually.be.fulfilled
   })
 
-  it('should not give back previously destroyed cart', function () {
+  it('should not give back previously destroyed cart', async function () {
+    storageMock.expects('get').once().returns(cartId)
+    await subjectUnderTest.load().should.eventually.be.fulfilled.and.be.equal(null)
+  })
+
+  it('should calculate the grand total of the cart correctly', async () => {
+    storageMock.expects('set').once().callsFake((cartIdKey, cartIdValue) => { cartId = cartIdValue })
+    await subjectUnderTest.addItems([BigCommerceCartRepository.createLineItem(112, 2)])
```

Comment:
It is the same situation here async function returns a promise that may be rejected - we should check if mocha will handle it properly. (this is what chai-as-promissed was supposed to do, therefore the concern)

Reply:
I made small modification so it will handle rejects properly